docs/guides/tutorials-appium/appium-cross-platform-automation.md

```diff
@@ -0,0 +1,159 @@
+
+
+###Steps
+1. Open your code Editor, create a project and write your [Test Setup](#test_setup) and your testing code using the [PageObject design pattern](#page_object) way and using [Appium annotations](#appium_anno).
+2. Open <a href="https://app.testobject.com" target="blank">TestObject</a> website, login and create two projects: the first project will be for your Android app and will contain your apk file, and the second project will be for your iOS app and will contain your ipa file.
+3. Create a suite in each project, and don't forget to add some devices to each suite (at least one).
+4. [Run](#run) your cross-platform test.
+5. <a href="#example" target="blank">Example</a>.
+
+
+<h3 id="page_object">PageObject design pattern</h3>
+Firstly we will start with PageObject design pattern which is a way of organizing the code you write in a certain structure. That means your code will be shorter, more readable and the user interaction flow will be evident. This way of coding encapsulates your code to make the proccess clearer and easier.
+
+A good example for using PageObject design pattern will be transformaing this test method from the normal way:
+
+{% highlight java %}
+@Test
+public void twoPlusTwoOperation() {
+    // Get elements from your app screen.
+    MobileElement buttonTwo = (MobileElement)(driver.findElement(By.id("net.ludeke.calculator:id/digit2")));
+    MobileElement buttonPlus = (MobileElement)(driver.findElement(By.id("net.ludeke.calculator:id/plus")));
+    MobileElement buttonEquals = (MobileElement)(driver.findElement(By.id("net.ludeke.calculator:id/equal")));
+    MobileElement resultField = (MobileElement)(driver.findElement(By.xpath("//android.widget.EditText[1]")));
+
+    // Add two and two ( 2 + 2 ).
+    buttonTwo.click();
+    buttonPlus.click();
+    buttonTwo.click();
+    buttonEquals.click();
+
+    // Check if within given time the correct result appears in the designated field ( check if 2 + 2 = 4 ).
+    new WebDriverWait(driver, 30).until(ExpectedConditions.textToBePresentInElement(resultField, 4));
+}
+{% endhighlight %}
+
+To the PageObject design pattern way:
+
+{% highlight java %}
+@Test
+public void twoPlusTwoOperation() {
+    app.calculatorScreen().addTwoAndTwo();
+    assertTrue(app.calculatorScreen().isResultCorrect("4"));
+}
+{% endhighlight %}
+
+For more information about PageObject, see our [Appium setup with PageObject](/docs/guides/appium-advanced-setup/) guide.
+
+<h3 id="appium_anno">Appium annotations</h3>
+It is hard to use the old way of finding elements for a cross-platform project, so we will use <strong>@AndroidFindBy()</strong> and <strong>@iOSFindBy()</strong> Appium annotations to fetch UI elements in our app, using (id, xpath or text, ...). Here in the folowing example we are tring to find an element using its Id on Android:
```

Comment:
It is probably not "hard", let's say there is a more convenient way.
"following", "trying"